scipy/spatial/transform/_rotation_groups.py

```diff
@@ -0,0 +1,90 @@
+import re
+import itertools
+import numpy as np
+from scipy.constants import golden as phi
+import scipy.spatial.transform
+
+
+def _icosahedral():
+    def even_permutations(x):
+        n = len(x)
+        m = np.zeros((n, n)).astype(np.int)
+        for p in itertools.permutations(x):
+            p = list(p)
+            m[:, :] = 0
+            m[np.arange(n), p] = 1
+            if np.linalg.det(m) > 0:
+                yield p
+
+    generators = np.array([[0.5, 0.5 / phi, 0.5 * phi, 0],
+                           [0.5, 0.5 / phi, -0.5 * phi, 0],
+                           [0.5, -0.5 / phi, 0.5 * phi, 0],
+                           [0.5, -0.5 / phi, -0.5 * phi, 0]])
+
+    g1 = _tetrahedral().as_quat()
+    g2 = np.array([e[p] for p in even_permutations(range(4))
+                   for e in generators])
+    return scipy.spatial.transform.Rotation.from_quat(np.concatenate((g1, g2)))
+
+
+def _octahedral():
+    g1 = _tetrahedral().as_rotvec()
+    g2 = np.pi / np.sqrt(2) * np.array([[1, 1, 0], [1, 0, 1],
+                                        [0, 1, 1], [1, -1, 0],
+                                        [1, 0, -1], [0, 1, -1]])
+    g3 = np.pi / 2 * np.eye(3)
+    return scipy.spatial.transform.Rotation.from_rotvec(np.concatenate((g1, g2, g3, -g3)))
+
+
+def _tetrahedral():
+    g1 = np.concatenate(([np.zeros(3)], np.pi * np.eye(3)))
+    g2 = np.array(list(itertools.product([-1., 1.], repeat=3)))
+    g2 *= 2 * np.pi / (3 * np.sqrt(3))
+    return scipy.spatial.transform.Rotation.from_rotvec(np.concatenate((g1, g2)))
+
+
+def _dicyclic(n, axis=2):
+    g1 = _cyclic(n, axis).as_rotvec()
+
+    thetas = np.linspace(0, np.pi, n, endpoint=False)
+    rv = np.pi * np.vstack([np.zeros(n), np.cos(thetas), np.sin(thetas)]).T
+    g2 = np.roll(rv, axis, axis=1)
+    return scipy.spatial.transform.Rotation.from_rotvec(np.concatenate((g1, g2)))
+
+
+def _cyclic(n, axis=2):
+    thetas = np.linspace(0, 2 * np.pi, n, endpoint=False)
+    rv = np.vstack([thetas, np.zeros(n), np.zeros(n)]).T
+    return scipy.spatial.transform.Rotation.from_rotvec(np.roll(rv, axis, axis=1))
+
+
+def _create_group(cls, group, axis='Z'):
```

Comment:
I think it's better to remove the underscore, because all the functions already reside inside a private module.

As for `_cyclic, etc.`, it makes more sense to keep the underscores, but I think it's fine to remove them as well.

Reply:
I have removed them.